src/raven.js

```diff
@@ -843,6 +843,38 @@ function makeRequest(opts) {
     img.src = src;
 }
 
+function makeXhrRequest(opts) {
+    var request = new XMLHttpRequest();
+    request.onreadystatechange = function (e) {
+        if (request.readyState !== 4) {
+            return;
+        }
+
+        if (request.status === 200) {
+            if (opts.onSuccess) {
+                opts.onSuccess();
+            }
+        } else {
+            if (opts.onError) {
+                opts.onError();
+            }
+        }
+    };
+
+    // NOTE: auth is intentionally sent as part of query string (NOT as custom
+    //       HTTP header) so as to avoid preflight CORS requests
+    request.open('POST', opts.url + '?' + urlencode(opts.auth));
+    request.send(JSON.stringify(opts.data));
+}
+
+function makeRequest(opts) {
+    var hasCORS =
+        'withCredentials' in new XMLHttpRequest() ||
+        typeof XDomainRequest !== 'undefined';
```

Comment:
I'm not an expert here, so lemme get this straight:

* You're checking for either `XMLHttpRequest` that has the proper support. Is `XMLHttpRequest` guaranteed to exist? I assume if it's undefined, we'll have an issue calling `new XMLHttpRequest()`
* Then falling back to checking for `XDomainRequest`, which is IE specific. This is great, but then inside `makeXhrRequest`, `XDomainRequest` isn't used or even attempted to be used. So it's not clear to me what good this check is doing.